### PR TITLE
Encapsulation of K_point modules

### PR DESCRIPTION
**Pull request by jwsyzy**
### What's changed?
- Move ::nks,nkstot,nkstot_ibz,nkstot_full:: to private
- Add related functions to set and get the above variables
- Add annotation to klist.cpp and klist.h

### CODE REVIEW

File: replace_nks.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
```
Comment (mohanchen) [inline replace_nks.sh]: we may not need this file in the current directory

File: source/module_cell/test/klist_test.cpp [hunk 2]
```diff
@@ -634,9 +634,9 @@ TEST_F(KlistTest, PrintKlistsWarnigQuit)
 TEST_F(KlistTest, SetBothKvecFinalSCF)
 {
 	kv->nspin = 1;
-	kv->nkstot = 1;
-	kv->nks = 1;
-	kv->renew(kv->nkstot);
+	kv->get_nkstot() = 1;
```
Comment (mohanchen) [inline source/module_cell/test/klist_test.cpp]: this replacement seems unreasonable

Comment (mohanchen) [reply source/module_cell/test/klist_test.cpp]: this replacement seems unreasonable

**Review by mohanchen — COMMENTED** [review]
some replacements seem unreasonable

File: source/module_cell/test/klist_test.cpp (hunk 2 above)
Comment (jwsyzy) [reply source/module_cell/test/klist_test.cpp]: Yes, we have modified it

File: replace_nks.sh (hunk 1 above)
Comment (jwsyzy) [reply replace_nks.sh]: ok, we have deleted it

File: source/module_elecstate/module_dm/test/test_cal_dm_R.cpp [hunk 3]
```diff
@@ -4,6 +4,51 @@
...
+#include "module_cell/klist.h"
+
+K_Vectors::K_Vectors()
+{
+}
+
+K_Vectors::~K_Vectors()
+{
+}
+int K_Vectors::get_nks() const
```
Comment (mohanchen) [inline source/module_elecstate/module_dm/test/test_cal_dm_R.cpp]: Could you directly define get_nks() and related functions in klist.h?

Comment (jwsyzy) [reply source/module_elecstate/module_dm/test/test_cal_dm_R.cpp]: Do you mean that we define the function directly in klist.h instead of klist.cpp?

**Review by mohanchen — APPROVED** [review]

**Pull request merged**